storagestate/storagestate_test.go

```diff
@@ -0,0 +1,1490 @@
+					Instance:  1,
+				},
+			},
+		},
+		{
+			SetupParams: storagestate.SetupParams{
+				InstanceIdent: cloudprotocol.InstanceIdent{
+					ServiceID: "service1",
+					SubjectID: "subject1",
+					Instance:  1,
+				},
+				UID:          1003,
+				GID:          1003,
+				StateQuota:   7,
+				StorageQuota: 1000,
+			},
+			instanceID: "instance1",
+			updateState: cloudprotocol.UpdateState{
+				InstanceIdent: cloudprotocol.InstanceIdent{
+					ServiceID: "service1",
+					SubjectID: "subject1",
+					Instance:  1,
+				},
+				State: "new state",
+			},
+		},
+		{
+			SetupParams: storagestate.SetupParams{
+				InstanceIdent: cloudprotocol.InstanceIdent{
+					ServiceID: "service1",
+					SubjectID: "subject1",
+					Instance:  1,
+				},
+				UID:          1003,
+				GID:          1003,
+				StateQuota:   7,
+				StorageQuota: 1000,
+			},
+			instanceID: "instance1",
+			updateState: cloudprotocol.UpdateState{
+				InstanceIdent: cloudprotocol.InstanceIdent{
+					ServiceID: "service1",
+					SubjectID: "subject1",
+					Instance:  1,
+				},
+				State:    "state",
+				Checksum: "decodingError",
+			},
+		},
+		{
+			SetupParams: storagestate.SetupParams{
+				InstanceIdent: cloudprotocol.InstanceIdent{
+					ServiceID: "service1",
+					SubjectID: "subject1",
+					Instance:  1,
+				},
+				UID:          1003,
+				GID:          1003,
+				StateQuota:   7,
+				StorageQuota: 1000,
+			},
+			instanceID: "instance1",
+			updateState: cloudprotocol.UpdateState{
+				InstanceIdent: cloudprotocol.InstanceIdent{
+					ServiceID: "service1",
+					SubjectID: "subject1",
+					Instance:  1,
+				},
+				State:    "state",
+				Checksum: hex.EncodeToString([]byte("bad checksum")),
+			},
+		},
+		{
+			SetupParams: storagestate.SetupParams{
+				InstanceIdent: cloudprotocol.InstanceIdent{
+					ServiceID: "service1",
+					SubjectID: "subject1",
+					Instance:  1,
+				},
+				UID:          1003,
+				GID:          1003,
+				StateQuota:   7,
+				StorageQuota: 1000,
+			},
+			instanceID:        "instance1",
+			errorSaveCheckSum: true,
+			updateState: cloudprotocol.UpdateState{
+				InstanceIdent: cloudprotocol.InstanceIdent{
+					ServiceID: "service1",
+					SubjectID: "subject1",
+					Instance:  1,
+				},
+				State:    "state",
+				Checksum: hex.EncodeToString(sumByte[:]),
+			},
+		},
+	}
+
+	storage := testStorageInterface{
+		data: make(map[string]storagestate.StorageStateInstanceInfo),
+	}
+
+	instance, err := storagestate.New(storageDir, stateDir, &storage)
+	if err != nil {
+		t.Fatalf("Can't create storagestate instance: %v", err)
+	}
+	defer instance.Close()
+
+	for _, testData := range testsData {
+		storage.errorSaveCheckSum = testData.errorSaveCheckSum
+
+		_, _, _, err := instance.Setup(testData.instanceID, testData.SetupParams)
+		if err != nil {
+			t.Fatalf("Can't setup instance: %v", err)
+		}
+
+		if err = instance.UpdateState(testData.updateState); err == nil {
+			t.Fatalf("State should not be updated")
```

Comment:
t.Fatal

Reply:
Done. But strange why linter doesn't handle this